packages/gatsby/src/redux/reducers/jobsv2.ts

```diff
@@ -11,22 +16,30 @@ export const jobsV2Reducer = (
     case `CREATE_JOB_V2`: {
       const { job, plugin } = action.payload
 
-      if (!job) return state
+      if (!job) {
+        throw new Error(
+          `If you encounter this error, it's probably a Gatsby internal bug. Please open an issue reporting us this.`
+        )
+      }
```

Comment:
This doesn't need a check, typings says it's not nullable so that's 👍 We'll have to handle this cause inside the redux/actions/public.js
```suggestion

```

Reply:
Nice! I've committed the changes directly from your suggestion.